test/test_transform.cpp

```diff
@@ -2074,4 +2074,313 @@ TEST(Transform_ArrayEraseVsArrayErase)
+    Associativity assoc{test_context, 2, changeset_dump_dir_gen.get()};
+    assoc.for_each_permutation([&](auto& it) {
+        auto server = &*it.server;
+        auto client_1 = &*it.clients[0];
+        auto client_2 = &*it.clients[1];
+
+        // Create baseline
+        client_1->transaction([&](Peer& c) {
+            auto& tr = *c.group;
+            auto table = tr.add_table_with_primary_key("class_Table", type_Int, "id");
+            table->add_column(type_Int, "int");
+        });
+
+        it.sync_all();
+
+        client_1->transaction([&](Peer& c) {
+            auto& tr = *c.group;
+            auto table = tr.get_table("class_Table");
+            auto obj = table->create_object_with_primary_key(123);
+            obj.set("int", 500);
+            obj.remove();
+        });
+
+        client_2->history.advance_time(1);
+        client_2->transaction([&](Peer& c) {
+            auto& tr = *c.group;
+            auto table = tr.get_table("class_Table");
+            table->create_object_with_primary_key(123);
+        });
+
+        it.sync_all();
+
+        ReadTransaction rt{server->shared_group};
+        auto table = rt.get_table("class_Table");
+        CHECK_EQUAL(table->size(), 0);
+    });
+}
+
+TEST(Transform_RSYNC_143_Fallout)
+{
+    // Divergence between Create-Set-Erase and Create.
+
+    auto changeset_dump_dir_gen = get_changeset_dump_dir_generator(test_context);
+    Associativity assoc{test_context, 2, changeset_dump_dir_gen.get()};
+    assoc.for_each_permutation([&](auto& it) {
+        auto server = &*it.server;
+        auto client_1 = &*it.clients[0];
+        auto client_2 = &*it.clients[1];
+
+        // Create baseline
+        client_1->transaction([&](Peer& c) {
+            auto& tr = *c.group;
+            auto table = tr.add_table_with_primary_key("class_Table", type_Int, "id");
+            table->add_column(type_Int, "int");
+        });
+
+        it.sync_all();
+
+        client_1->transaction([&](Peer& c) {
+            auto& tr = *c.group;
+            auto table = tr.get_table("class_Table");
+            auto obj = table->create_object_with_primary_key(123);
+            obj.set("int", 500);
+        });
+
+        it.sync_all();
+
+        client_1->history.advance_time(1);
+        client_1->transaction([&](Peer& c) {
+            auto& tr = *c.group;
+            auto table = tr.get_table("class_Table");
+            auto obj = table->get_object_with_primary_key(123);
+            obj.remove();
+        });
+
+        client_2->history.advance_time(1);
+        client_2->transaction([&](Peer& c) {
+            auto& tr = *c.group;
+            auto table = tr.get_table("class_Table");
+            auto obj = table->create_object_with_primary_key(123);
+            obj.set("int", 900);
+            obj.remove();
+        });
+
+        it.sync_all();
+
+        static_cast<void>(server);
+    });
+}
+
+TEST(Transform_ArrayClearVersusClearRegression)
```

Comment:
Can you add a comment here explaining what this test is testing. Seems like the name doesnt really give a sense of what is being tested and it can be hard to follow along what is going on in this large test

Reply:
Sure thing